### PR TITLE
feat: conditionally load focus-visible polyfill

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
Test perf benefits of only loading the polyfill when needed.
- use dynamic import and capability test to prevent polyfill use in Chrome and Firefox
- ungated usage can cause edge case situations like in #2071 

## Motivation and context
Perf

## How has this been tested?

-   [ ] _Test case 1_
    1. Go [here](https://focus-visible--spectrum-web-components.netlify.app/)
    2. Navigate the site with the keyboard and see that focus visible states are triggered.

## Types of changes
-   [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

**Comment by benjamind** [comment]
Interesting how much this affects popover performance. I wonder if the polyfill was what was causing the excessive focus event interactions I was seeing on my FocusFollower experiments we were discussing yesterday?

**Comment by Westbrook** [comment]
This includes `popover` in the perf tests only because `popover` depends of `overlay` which depends on `shared` where this code was changes, but doesn't include it itself... so the variance of `14.36ms - 17.44ms` for 100 instances of the `popover` element is likely more noise than anything else.

For the added focuses, I'd less think that was the polyfill that the handling for certain elements within the library. If the lack of the polyfill did remove them somehow, then winning!

File: packages/shared/src/focus-visible.ts [hunk 1]
```diff
@@ -40,6 +38,9 @@ try {
     document.body.querySelector(':focus-visible');
 } catch (error) {
     hasFocusVisible = false;
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    import('focus-visible');
```
Comment (Westbrook) [inline packages/shared/src/focus-visible.ts:43]: This is the guts of this PR. Everything else is test support for this change in browsers that do not require the polyfill, which is ALL of them now! 🥳

**Review by najikahalsema — APPROVED** [review]
yay!!!
